### DISCUSSION TITLE
rtpengine?

### DISCUSSION THREAD

**Discussion by jailbird777** [hnimminh/libresbc#50] [Q&A] [answered]
I'm currently using sbcOS, which is nice but pretty much unmaintained. So I was looking at switching to LibreSBC.

I was curious if there was a reason that you went with a full FreeSWITCH installation instead of using Kamailio's rtpengine module along with rtpengine? Since rtpengine can off-load to the kernel, that seems like it would be more performant?

**Comment by hnimminh** [comment, marked as answer]
Hi @jailbird777 thanks for interest to LibreSBC project, 

Theoretically, SBC should work as a B2BUA, The Kamailio is naturally Proxy architecture seem not a good fix for SBC (even yes, kamailio can be SBC) here is some of reason:

- As proxy, Kamailio work like garbage in & garbage out
- Un-True topology hiding
- Difficult to support multiple network interface

Yes, Performance is very important, Basically, on the same hardware Kamailio + Rtpengine beat the FreeSWITCH. My [load test](https://github.com/hnimminh/libresbc/wiki) show LibreSBC performance is good enough for single instance, and in the high load we can scale libreSBC to run in cluster mode with multiple instances to improve performance. 

Hope this answer you concern!

**Reply by jailbird777** [reply]
Makes sense, thanks!
